data.go

```diff
@@ -118,7 +118,7 @@ func (s *SessionManager) Commit(ctx context.Context) (string, time.Time, error)
 
 	expiry := sd.deadline
 	if s.IdleTimeout > 0 {
-		ie := time.Now().Add(s.IdleTimeout)
+		ie := time.Now().Add(s.IdleTimeout).UTC()
 		if ie.Before(expiry) {
```

Comment:
Is this really correct. Shouldn't it be `After`?

Reply:
I think it should be `Before`. We only want to update the expiry time to `now + idle timeout` if that is before the fixed end of life on the session. Or in other words, we want the session to expire on whichever is earlier: `now + idle timeout` or `creation time + session lifetime`.

Good spot on the UTC part, thanks!

Reply:
But in such case it will never increase. Let's say in situation where (Lifetime is 1d and we set IdelTimeout 20min).

Let's see sample scenario:
```
[00:00:00] Req 1 - `create session` - deadline = now+1d
                          - `commit` - now + 20min < deadline -> so deadline = now + 20min

[00:10:00] Req 2 - `commit` - now + 20min > deadline -> so deadline is not changed

[00:20:00] Req 3 - `commit` - now + 20min > deadline -> so deadline is not changed

[00:25:00] Req 4 - session expired
```

Even if requests were made in less time than `IdleTimeout` session will still expire.

To be honest there is no much point in having two `Lifetime` and `IdleTimeout` settings. Single setting`IdleTimeout` (ususally default 20min) is enaugh and session `deadline` should be increased on every request

Reply:
IMHO this is how it should work https://github.com/lafriks/scs/commit/585fe036f066bae6a3d0e4b050e696006e39a7ab

It drops lifetime and and makes idle timeout to work as sliding expiration. Also not to update cookie expiration all the time it makes cookie as session cookie in browser (as this is also what most session managers in frameworks do).

Of course this is breaking change. Should I submit PR for that?

Reply:
Yes, the existing behavior is intended. What you're suggesting is dangerous because it means an attacker could keep a hijacked session alive and impersonate the user *forever*.

Please see https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html#absolute-timeout

Reply:
Both are equally dangerous same as 1d absolute timeout. Current `IdleTimeout` functionality is completely broken as per https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html#idle-timeout

Reply:
Can you explain what you mean by completely broken? As far as I can see, the current implementation does exactly what you have linked to.

Reply:
Just to make explicit: scs currently supports idle `IdleTimeout` and absolute timeout `Lifetime`. You can have both set on the same session. The absolute timeout is a fixed endpoint that never changes or can be extended. The IdleTimeout if enabled will be extended by the approriate duration with each request. Both are stored server side so can't be manipulated.

Reply:
I've just seen your sample scenario above. I think the problem is that you don't yet know that there is a difference between deadline and expiry. Deadline is stored as part of the session data in the session.Deadline field. It equals the original now+1d and never changes.

Reply:
`IdleTimeout` should be timeout after last request but currently it does work same as absolute timeout.